server/api.go

```diff
@@ -931,3 +942,124 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request) {
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getRepositories(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	ctx := context.Background()
+	githubClient := p.githubConnect(*info.Token)
+
+	repositories, _, apiErr := githubClient.Repositories.List(ctx, "", &github.RepositoryListOptions{})
```

Comment:
`RepositoryListOptions` used `Page` and `PerPage` for paging. Do you catch the case where there are more then repos then fit into a response? See also https://pkg.go.dev/github.com/google/go-github/v31/github?tab=doc#hdr-Pagination

Reply:
fixed in f4a1cc3